test/CadlRanchProjects.Tests/type-model-visibility.cs

```diff
@@ -46,5 +45,29 @@ public void ReadOnlyPropertiesAreReadOnly()
             Assert.Null(listProperty.SetMethod);
             Assert.AreEqual(typeof(IReadOnlyList<int>), listProperty.PropertyType);
         }
+
+        [Test]
+        public void RequiredPropertiesAreSetable()
+        {
+            var requiredInt = TypeAsserts.HasProperty(typeof(VisibilityModel), nameof(VisibilityModel.QueryProp), BindingFlags.Public | BindingFlags.Instance);
+            Assert.NotNull(requiredInt.SetMethod);
+        }
+
+        [Test]
+        public void RequiredListsAreReadOnly()
```

Comment:
it should be not settable, readonly has ambiguities. 
```suggestion
        public void RequiredListsAreNotSettable()
```

Reply:
fixed